src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/utils/DateConversionHelper.kt

```diff
@@ -20,9 +20,9 @@ class DateConversionHelper {
         "dd/MM/yyyy",
         "dd MMMM yyyy",
       ),
-      // "2024-05-01T12:34:56[.1|12|123|1234|12345|123456][Z][+00:00]"
+      // "2024-05-01T12:34:56[.1|12|123|1234|12345|123456|1234567|12345678|123456789][Z][+00:00]"
       DateConversion(
-        "^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}(\\.\\d{1,6})?Z?(\\+\\d{2}:\\d{2})?$".toRegex(),
+        "^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}(\\.\\d{1,9})?Z?(\\+\\d{2}:\\d{2})?$".toRegex(),
```

Comment:
Needed this change due to one of the date values received for approved premises

Reply:
is this the milliseconds/nano part of the date time? Regex hurts my brain at the best of times

Reply:
yep, increasing to 9 digits allowed for that but